builder/remotecontext/git/gitutils.go

```diff
@@ -70,7 +70,7 @@ func fetchArgs(remoteURL *url.URL, ref string) []string {
 	shallow := true
 
 	if strings.HasPrefix(remoteURL.Scheme, "http") {
-		res, err := http.Head(fmt.Sprintf("%s/info/refs?service=git-upload-pack", remoteURL))
+		res, err := http.Get(fmt.Sprintf("%s/info/refs?service=git-upload-pack", remoteURL))
```

Comment:
The `HEAD` and fallback sounds reasonable, but it doesn't solve the special case with GitHub, as GitHub doesn't support `HEAD`. In the interest of performance, we may just want to do no requests at all with GitHub.